src/util/event_emitter.js

```diff
@@ -0,0 +1,140 @@
+export class EventEmitter {
+  constructor() {
+    this._handlers = [];
+  }
+
+  /**
+   * Adds the event name and handler function to the end of the handlers array.
+   * No checks are made to see if the handler has already been added.
+   * Multiple calls passing the same combination of event name and handler will result in the handler being added,
+   * and called, multiple times.
+   * @param {String} event
+   * @param {Function} handler
+   * @returns {EventEmitter}
+   */
+  on(event, handler) {
+    this._handlers.push({
+      event,
+      handler
+    });
+
+    return this;
+  }
+
+  _onceWrapper(...args) {
```

Comment:
onceWrapper don't need to be a class method, it should be a normal function, or private function inside `_onceWrap`

Reply:
I'll move these functions out